web/src/lib/utils/asset-utils.ts

```diff
@@ -513,7 +513,7 @@ export const selectAllAssets = async (timelineManager: TimelineManager, assetInt
 
   try {
     for (const monthGroup of timelineManager.months) {
-      await timelineManager.loadMonthGroup(monthGroup.yearMonth);
+      await timelineManager.loadSegment(monthGroup.identifier);
```

Comment:
You are iterating over a property on the manager and then calling a method on the manager with a property of the iterated value, which then internally _looks up the original month_ and does something with it. This abstraction makes no sense. Loop over segments and call `loadSegment` with the current segment.